includes/reports/class-sensei-no-users-table-relationship.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * File containing the Sensei_No_Users_Table_Relationship class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class that handles a special case of environments that have the users table in a separate
+ * database from the other tables.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_No_Users_Table_Relationship {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_No_Users_Table_Relationship constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		// Students report.
+		add_filter( 'sensei_students_report_last_activity_filter_enabled', [ $this, 'can_use_users_relationship' ] );
+		add_filter( 'sensei_analysis_overview_users_columns_sortable', [ $this, 'filter_analysis_overview_users_columns_sortable' ] );
+		add_filter( 'sensei_analysis_overview_column_data', [ $this, 'filter_analysis_overview_column_data' ], 10, 3 );
+	}
+
+	/**
+	 * Check if it's possible to use a relationship between users and posts table. It's used to
+	 * check environments where the users table lives in a different places and the relationship
+	 * doesn't work.
+	 *
+	 * @access private
+	 *
+	 * @return boolean Whether the users relationship is possible.
+	 */
+	public function can_use_users_relationship() {
+		/**
+		 * Filters if site environment is able make queries including a relationship between users
+		 * table and others.
+		 *
+		 * @hook  sensei_can_use_users_relationship
+		 * @since $$next-version$$
+		 *
+		 * @param {null} $can_use_users_relationship Default value is `null`. With this value the
+		 *                                           filter is ignored.
+		 *
+		 * @return {boolean|null} Whether the site environment is able to make queries including a
+		 *                        relationship between users table and others..
+		 */
+		$filtered_can_use_users_relationship = apply_filters( 'sensei_can_use_users_relationship', null );
```

Comment:
It's not used yet, but it's here to be used in case of any particular situation of a specific environment.